stackit/internal/services/serverbackup/schedule/resource.go

```diff
@@ -0,0 +1,539 @@
+package serverbackup
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/stackitcloud/terraform-provider-stackit/stackit/internal/conversion"
+	"github.com/stackitcloud/terraform-provider-stackit/stackit/internal/core"
+	"github.com/stackitcloud/terraform-provider-stackit/stackit/internal/validate"
+
+	"github.com/stackitcloud/stackit-sdk-go/core/config"
+	"github.com/stackitcloud/stackit-sdk-go/core/oapierror"
+	"github.com/stackitcloud/stackit-sdk-go/services/serverbackup"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &scheduleResource{}
+	_ resource.ResourceWithConfigure   = &scheduleResource{}
+	_ resource.ResourceWithImportState = &scheduleResource{}
+)
+
+type scheduleResourceModel struct {
+	ID               types.String                   `tfsdk:"id"`
+	ProjectId        types.String                   `tfsdk:"project_id"`
+	ServerId         types.String                   `tfsdk:"server_id"`
+	BackupScheduleId types.Int64                    `tfsdk:"backup_schedule_id"`
+	Name             types.String                   `tfsdk:"name"`
+	Rrule            types.String                   `tfsdk:"rrule"`
+	Enabled          types.Bool                     `tfsdk:"enabled"`
+	BackupProperties *scheduleBackupPropertiesModel `tfsdk:"backup_properties"`
+}
+
+// NewScheduleResource is a helper function to simplify the provider implementation.
+func NewScheduleResource() resource.Resource {
+	return &scheduleResource{}
+}
+
+// scheduleResource is the resource implementation.
+type scheduleResource struct {
+	client *serverbackup.APIClient
+}
+
+// Metadata returns the resource type name.
+func (r *scheduleResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_serverbackup_schedule"
```

Comment:
I would prefer `stackit_server_backup_schedule`, what do you think?

Reply:
ok